buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -0,0 +1,65 @@
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.findByType
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.targets.jvm.KotlinJvmTarget
+import java.util.spi.ToolProvider
+
+object Java9Modularity {
+    @JvmStatic
+    fun Project.configureJava9ModuleInfo(moduleName: String) {
+        val multiplatform = extensions.findByType<KotlinMultiplatformExtension>() ?: return
```

Comment:
Will it work with `Hocon` format that is JVM-only?

Reply:
No, I deliberately left modularity support out for the `Hocon` format because the structure of that (sub)project was quite different from the other formats. But it does make sense to add JPMS support for `Hocon` too.